bgapi/module.py

```diff
@@ -207,7 +208,18 @@ def get_services(self):
     def get_characteristics(self):
         return [j for i, j in sorted(self.characteristics.items(), key=operator.itemgetter(0))]
 
+    def get_gatt_database(self):
+        return self.gatt_database
+
     def update_uuid(self, handle, uuid):
+        # TODO: I am sure there is a better way to do this
+        uuid_hex_val = 0
+        for i in uuid[::-1]:
+            uuid_hex_val <<= 8
+            uuid_hex_val |= ord(i)
+        handle_dict = {"Handle" : handle, "UUID" : uuid_hex_val}
+        if handle_dict not in self.gatt_database:
+            self.gatt_database.append({"Handle" : handle, "UUID" : uuid_hex_val})
```

Comment:
I don't like the names I used, "handle_uuid" and "uuid_handle" isn't nearly descriptive enough. Those dictionaries can go away.  I like the descriptiveness of yours, but it relies on the "walk gatt" for ordering.  It would be cooler if the self.gatt_database list were by sorted by handle.  i.e. self.gatt_database[0] = {"Handle": 0, "UUID": some_hex}.  This would mean making self.gatt_database a dictionary instead of a list.